lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -704,6 +704,10 @@ public function add($dqlPartName, $dqlPart, $append = false)
             }
 
             $dqlPart = $newDqlPart;
+
+            if (!$append) {
+                $isMultiple = false;
```

Comment:
@Ocramius Do you propose a different solution that will make it more clear to understand?

Like this:
```php
$isMultiple = is_array($this->_dqlParts[$dqlPartName]) && $dqlPartName != 'join';
```